app/process_proposal.go

```diff
@@ -82,6 +86,49 @@ func (app *App) ProcessProposal(req abci.RequestProcessProposal) abci.ResponsePr
 		}
 	}
 
+	data, err := coretypes.DataFromProto(req.BlockData)
+	if err != nil {
+		app.Logger().Error(rejectedPropBlockLog, "reason", "failure to unmarshal block data:", err)
+		return abci.ResponseProcessProposal{
+			Result: abci.ResponseProcessProposal_REJECT,
+		}
+	}
+
+	shares, _, err := data.ComputeShares(req.BlockData.OriginalSquareSize)
+	if err != nil {
+		app.Logger().Error(rejectedPropBlockLog, "reason", "failure to compute shares from block data:", err)
```

Comment:
how does this look like when printed? are the extra `:` needed?

Reply:
this exact error is difficult to reproduce, but they look like this

```
E[2022-05-11|20:46:44.403] Rejected proposal block:                     reason="missing MsgPayForMessage for included message" error="tacos are good on other days than tuesday"
```
I was however missing an error key, which has been added